libs/wingsdk/src/target-tf-aws/service.ts

```diff
@@ -0,0 +1,338 @@
+import { AwsInflightHost, IAwsInflightHost, NetworkConfig, PolicyStatement } from "../shared-aws";
+import { EcsTaskDefinition } from "../.gen/providers/aws/ecs-task-definition";
+import { EcsService } from "../.gen/providers/aws/ecs-service";
+import { RegistryImage } from "../.gen/providers/docker/registry-image";
+import { IamRole } from "../.gen/providers/aws/iam-role";
+import { SecurityGroup } from "../.gen/providers/aws/security-group";
+import { Image } from "../.gen/providers/docker/image";
+import { CloudwatchLogGroup } from "../.gen/providers/aws/cloudwatch-log-group";
+import { IInflightHost } from "../std";
+import { Lazy } from "cdktf";
+import { LiftMap } from "../core";
+
+
+export class Service extends cloud.Service implements IAwsInflightHost {
+  private workdir: string;
+  private wrapperEntrypoint: string;
+  private policyStatments?: any[];
+  private dockerFileName: string;
+  private service: EcsService;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    handler: cloud.IServiceHandler,
+    props: cloud.ServiceProps = {}
+  ) {
+    super(scope, id, handler, props);
+    // We probably need to create the ECR per app (App.of(this).ecr)
+
+    this.workdir = App.of(this).workdir;
+    this.wrapperEntrypoint = join(this.workdir, `${this.assetName}_wrapper.js`);
+    this.dockerFileName = `Dockerfile_${this.assetName}`;
+    let app = App.of(this) as App;
+
+    // This forces the lazy loading of the properties (TODO: find a better way to do this)
+    app.ecrAuth;
+    app.dockerProvider;
+    app.vpc;
+    
+    let image = new Image(this, "DockerImage", {
+      name: `${app.ecr.repositoryUrl}:${this.assetName}`,
+      buildAttribute: {
+        context: ".wing",
+        dockerfile: this.dockerFileName,
+        platform: "linux/amd64",
+      }
+    });
+
+    new RegistryImage(this, "RegistryImage", {
+      name: image.name
+    });
+
+    let executionRole = new IamRole(this, "ExecutionRole", {
+      assumeRolePolicy: JSON.stringify({
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Effect: "Allow",
+            Principal: {
+              Service: "ecs-tasks.amazonaws.com",
+            },
+            Action: "sts:AssumeRole",
+          },
+        ],
+      }),
+      inlinePolicy: [{
+        name: `${this.assetName}-inline-policy`,
+        policy: JSON.stringify({
+          Version: "2012-10-17",
+          Statement: [
+            {
+              "Effect": "Allow",
+              "Action": [
+                "logs:CreateLogStream",
+                "logs:PutLogEvents",
+                "logs:CreateLogGroup"
+              ],
+              "Resource": "*"
+            },
+            {
+              "Effect": "Allow",
+              "Action": [
+                "ecr:BatchGetImage",
+                "ecr:GetDownloadUrlForLayer",
+                "ecr:GetAuthorizationToken"
+              ],
+              "Resource": "*"
```

Comment:
I think you should be able to be more specific with this since you are creating the image repository, no?